lib/git/lib.rb

```diff
@@ -458,11 +454,7 @@ def config_list
         parse_config_list command_lines('config', ['--list'])
       end
 
-      if @git_dir
-        Dir.chdir(@git_dir, &build_list)
-      else
-        build_list.call
-      end
+      build_list.call(@git_dir)
```

Comment:
same here, I think we need to utilize the `path` variable in the `config` command if it exists